src/Tools/DynamoInstallDetective/ProductLookUp.cs

```diff
@@ -202,8 +281,9 @@ public IInstalledProduct GetProductFromProductCode(string productCode)
 
         public virtual IEnumerable<string> GetProductNameList()
         {
-            var key = OpenKey(REG_KEY64);
-            return key.GetSubKeyNames().Where(s => s.Contains(ProductLookUpName));
+            return RegUtils.GetInstalledProducts().Select(s => s.Value.DisplayName).Where(s => {
+                return s?.Contains(ProductLookUpName) ?? false;
+            });
```

Comment:
Minor point but I was wondering if we could just have a property on `InstalledProductLookup` or `IProductLookup` that stored a list of installed products called `InstalledProduct`:
```
private List<string> installedProducts;
List<string> InstalledProducts 
{
    get 
    {
         if(installedProducts != null) return installedProducts;
         installedProducts = RegUtils.GetInstalledProducts();
         return installedProducts;
    }
}

Reply:
Then say this on line 284:
```
return InstalledProducts.Select(s => s.Value.DisplayName).Where(s => {
                return s?.Contains(ProductLookUpName) ?? false;
```

Reply:
hmm...this might work but I have some concerns:
I would change the code a bit to have the `installedProducts` as private static (static so that the installedProducts can be reused across instances of InstalledProductLookup )
I do not think we can add properties to Interfaces like IProductLookup .

The benefit of creating the cache in a specific scope is that we can erase the cache at the end. I don't know if we can control when a static property in the InstalledProductLookup would get cleaned up. The state of the installedProducts  data would stay the same for the rest of the process lifetime (maybe that is not an issue...). And I don;t think we can control when the static property can get cleaned up..

Reply:
I don't think it needs to be static as `InstalledProductLookup` is initialized only once, I think, at least for ASM lookup. This simplifies the implementation.

Reply:
In the example bellow :
`
foreach (var productSearchPattern in productSearchPatterns) { installs.LookUpAndInitProducts(new InstalledProductLookUp(productSearchPattern, fileSearchPattern)); }
`
We have multiple `InstalledProductLookup` instances. How do you mean `InstalledProductLookup` is initialized only once ?

Reply:
Oh, in that case, scratch that.

Reply:
Using it as static will make it work...but the only (biggest) downside I see is that the data in the static will live on for the entire lifetime of the process. If there are a lot of records in the windows registry, it will be useless data that Dynamo will carry around...